environments/azure/flyte-core/iam.tf

```diff
@@ -0,0 +1,89 @@
+
+data "azuread_client_config" "current" {}
+
```

Comment:
On the topic of using workload identities with AKS. I think there are two possibilities to enable this:
Service Principals(your approach) vs Managed Identities (I like these more).

[What the difference?](https://devblogs.microsoft.com/devops/demystifying-service-principals-managed-identities/)

On this one we can chat in private but i think going with Managed Identities is more save (no credentials) and requires less terraform code to setup.

Reply:
I originally went with the Managed Identities approach but the issue was that role assignment was only possible for a Service Principal and not a MI. The Flyte users who have Flyte running on AZ are using SPs. But I agree MI would be better. I'll give it a second try and will update here accordingly. Thanks!